### PR TITLE
Forum App Models, Views, Serializers, Urls

### PR DESCRIPTION
**Pull request by colleenobrien3**
- Related Issue (include '#'): #513 , #514 , #515 

- Description of changes made: app models for reply, comment, and post, also serializers, views, and urls

- Is the feature complete/bug resolved/etc..: may need to edit relationships once we have seed data and once we have other apps merged

- Any known bugs/strange behavior:

- Is there specific feedback you would like on these changes:

- Screenshot(s):

### CODE REVIEW

File: api/forum/models.py [hunk 1]
```diff
@@ -1,3 +1,40 @@
 from django.db import models
+from accounts.models import Profile
 
-# Create your models here.
+
+class Post(models.Model):
```
Comment (ZakkMan) [inline api/forum/models.py:5]: These models are all largely the same, is there a way to combine them into one?

For instance, could you just have a `Post` model with a field like `parent` that relates one Post instance to another Post instance?

File: api/forum/models.py [hunk 2]
```diff
@@ -1,3 +1,16 @@
...
+    parent = models.ForeignKey('self', related_name='comments', on_delete=models.CASCADE, null=True, blank=True, default=1)
```
Comment (ZakkMan) [inline api/forum/models.py:6]: With just the `ForeignKey` I think you'll only be able to have one 'comment'. I think you'll need a many-to-many.

File: api/forum/models.py [hunk 3]
```diff
@@ -1,3 +1,16 @@
...
+    post_type = models.CharField(max_length=200, null=True, blank=True)
```
Comment (ZakkMan) [inline api/forum/models.py:7]: What is this field for?

**Comment by colleenobrien3** [comment]
I made it a manytomany and tested it in the django ui. A post can have multiple comments now. Also, I added the type field to differentiate comments from posts and replies. Each type of post has different styling, according to the figma, so I thought we needed some way of differentiating them. @ZakkMan

**Comment by ZakkMan** [comment]
@colleenobrien3 can you update this with develop and then merge it?

**Pull request merged**